src/devices/bus/heathzenith/h8/wh_8_64.cpp

```diff
@@ -0,0 +1,314 @@
+	PORT_DIPSETTING(    0x80, DEF_STR( On ) )
+
+INPUT_PORTS_END
+#endif
+
+wh_8_64_48k_device::wh_8_64_48k_device(const machine_config &mconfig, const char *tag, device_t *owner, u32 clock):
+	wh_8_64_device(mconfig, H8BUS_WH_8_64_48K, tag, owner, 0)
+{
+}
+
+void wh_8_64_48k_device::device_reset()
+{
+	if (!m_installed)
+	{
+		h8bus().space(AS_PROGRAM).install_readwrite_handler(0x2000, 0x3fff,
+			read8sm_delegate(m_ram[0], FUNC(ram_device::read)),
+			write8sm_delegate(m_ram[0], FUNC(ram_device::write)));
+
+		h8bus().space(AS_PROGRAM).install_readwrite_handler(0x4000, 0x5fff,
+			read8sm_delegate(m_ram[1], FUNC(ram_device::read)),
+			write8sm_delegate(m_ram[1], FUNC(ram_device::write)));
+
+		h8bus().space(AS_PROGRAM).install_readwrite_handler(0x6000, 0x7fff,
+			read8sm_delegate(m_ram[2], FUNC(ram_device::read)),
+			write8sm_delegate(m_ram[2], FUNC(ram_device::write)));
+
+		h8bus().space(AS_PROGRAM).install_readwrite_handler(0x8000, 0x9fff,
+			read8sm_delegate(m_ram[3], FUNC(ram_device::read)),
+			write8sm_delegate(m_ram[3], FUNC(ram_device::write)));
+
+		h8bus().space(AS_PROGRAM).install_readwrite_handler(0xa000, 0xbfff,
+			read8sm_delegate(m_ram[4], FUNC(ram_device::read)),
+			write8sm_delegate(m_ram[4], FUNC(ram_device::write)));
+
+		h8bus().space(AS_PROGRAM).install_readwrite_handler(0xc000, 0xdfff,
+			read8sm_delegate(m_ram[5], FUNC(ram_device::read)),
+			write8sm_delegate(m_ram[5], FUNC(ram_device::write)));
+
+		m_installed = true;
+	}
+}
+
+
+wh_8_64_32k_device::wh_8_64_32k_device(const machine_config &mconfig, const char *tag, device_t *owner, u32 clock):
+	wh_8_64_device(mconfig, H8BUS_WH_8_64_32K, tag, owner, 0)
+{
+}
+
+void wh_8_64_32k_device::device_reset()
+{
+	if (!m_installed)
+	{
+		h8bus().space(AS_PROGRAM).install_readwrite_handler(0x2000, 0x3fff,
+			read8sm_delegate(m_ram[0], FUNC(ram_device::read)),
+			write8sm_delegate(m_ram[0], FUNC(ram_device::write)));
+
+		h8bus().space(AS_PROGRAM).install_readwrite_handler(0x4000, 0x5fff,
+			read8sm_delegate(m_ram[1], FUNC(ram_device::read)),
+			write8sm_delegate(m_ram[1], FUNC(ram_device::write)));
+
+		h8bus().space(AS_PROGRAM).install_readwrite_handler(0x6000, 0x7fff,
+			read8sm_delegate(m_ram[2], FUNC(ram_device::read)),
+			write8sm_delegate(m_ram[2], FUNC(ram_device::write)));
+
+		h8bus().space(AS_PROGRAM).install_readwrite_handler(0x8000, 0x9fff,
+			read8sm_delegate(m_ram[3], FUNC(ram_device::read)),
+			write8sm_delegate(m_ram[3], FUNC(ram_device::write)));
+
+		m_installed = true;
+	}
+}
+
+} // anonymous namespace
+
+DEFINE_DEVICE_TYPE_PRIVATE(H8BUS_WH_8_64,     device_h8bus_card_interface, wh_8_64_device,     "wh8_h_8_64",     "Heath WH-8-64 64k Dynamic RAM");
+DEFINE_DEVICE_TYPE_PRIVATE(H8BUS_WH_8_64_48K, device_h8bus_card_interface, wh_8_64_48k_device, "wh8_h_8_64_48k", "Heath WH-8-64 48k Dynamic RAM");
+DEFINE_DEVICE_TYPE_PRIVATE(H8BUS_WH_8_64_32K, device_h8bus_card_interface, wh_8_64_32k_device, "wh8_h_8_64_32k", "Heath WH-8-64 32k Dynamic RAM");
```

Comment:
Is it worth having separate card devices for the different RAM configurations, or would you be better off just having a machine configuration “switch” to set the amount of RAM present?  (For example the MDC4•8/MDC8•24 cards use a fake switch to select the video RAM size.)

Reply:
Based on this prior comment from @pmackinlay  - https://github.com/mamedev/mame/pull/13040#discussion_r1879628078 about how to handle the m_connected_tlbc, I split that card into 2 slot options. On the other PR it felt cleaner with the 2 cards, but I'm not sure which would be better with this board. And followed that with this PR. But if you want it handled by a machine config switch, I can change it.